README.md

```diff
@@ -1,15 +1,104 @@
 # trace-server-protocol
 
-Specification of the Trace Server Protocol
-
-This protocol is built to decouple the backend and frontend of trace analysers, allowing traces to reside and be analysed on the backend, and visual models to be exchanged with a variety of clients.
-
-The protocol is meant to be RESTful, over HTTP.
+Specification of the Trace Server Protocol (TSP).
 
 The specification is currently written in **OpenAPI 3.0** and can be pretty-visualized in the [github pages][tspGhPages].
 
 **👋 Want to help?** Read our [contributor guide][contributing].
 
+## About, why and goals of the TSP:
+
+Similarly to the philosophy behind the Language Server Protocol (LSP),
+the TSP is an open, HTTP-based protocol for use between *analytics or
+interactive visualization web applications* and *servers that provide
+trace analysis specific features* such as:
+
+- generation of data that can interpeted as charts from trace analysis
+- selection, filtering and correlation of elements in charts using data
+  contained in the traces
+- statistics calculation from data contained in the traces
+
+The goal of the protocol is to allow trace analysis support to be
+implemented and distributed independently of any given analytics or
+interactive visualization web application.
+
+## What the TSP is:
+
+The TSP is a RESTful API on top of the HTTP protocol that enables to:
+
+- analyze time series data (e.g. traces and logs);
```

Comment:
I think this has been one of the biggest sources of contention:
Is it "Analyze time series data (e.g. traces and logs)"
or
"Analyze trace and log data (E.g. CPU traces, GPU traces, software logs etc.)"

This will decide what is "domain-specific" (i.e. are Traces/Logs themselves a domain or are GPU traces a domain). Since tsp and the viewer can't be made directly aware of anything domain specific, this choice will dictate the kind of explicit end-points and required parameters we can have, v/s, what we need to pass only implicitly as optional parameters.